logging/logging.go

```diff
@@ -40,89 +38,100 @@ type Config struct {
 	Writer io.Writer
 }
 
+// Setup takes as an arugment a configuration and then uses it to configure the global
+// logger. After setup logging.Global() can be used to access the global logger
 func Setup(config *Config) error {
-	// Set the global log level, this will be used by all loggers
-	LogLevel = config.Level
 
-	logFilter, err := setupFilter(config.Writer)
-	if err != nil {
-		return fmt.Errorf("error setting up log filter: %s", err)
+	// Validate the log level
+	if !validateLogLevel(config.Level) {
+		return fmt.Errorf("Invalid log level: %s. Valid log levels are: %v",
+			config.Level,
+			Levels)
 	}
 
-	var logOutput io.Writer
+	// Set the global log level, this will be used by all loggers
+	setLogLevel(config.Level)
 
+	var logOutput io.Writer
 	// Check if syslog is enabled
 	if config.Syslog {
-		log.Printf("[INFO] (logging) enabling syslog on %s", config.SyslogFacility)
-
-		l, err := gsyslog.NewLogger(gsyslog.LOG_NOTICE, config.SyslogFacility, config.SyslogName)
+		syslog, err := gsyslog.NewLogger(gsyslog.LOG_NOTICE, config.SyslogFacility, config.SyslogName)
 		if err != nil {
 			return fmt.Errorf("error setting up syslog logger: %s", err)
 		}
-		syslog := &SyslogWrapper{l, logFilter}
-		logOutput = io.MultiWriter(logFilter, syslog)
+		logOutput = io.MultiWriter(config.Writer, syslog)
 	} else {
-		logOutput = io.MultiWriter(logFilter)
+		logOutput = config.Writer
 	}
 
-	log.SetFlags(log.Ldate | log.Ltime | log.Lmicroseconds | log.LUTC)
-	log.SetOutput(logOutput)
+	logger := hclog.New(&hclog.LoggerOptions{
+		Level:      hclog.LevelFromString(LogLevel),
+		Output:     logOutput,
+		TimeFormat: hclog.TimeFormat,
+	})
 
+	hclog.SetDefault(logger)
 	return nil
 }
 
+// Logger is a type alias for hclog.Logger
+type Logger hclog.Logger
```

Comment:
Although the Logger type is used below, it's a quicker to reference defined types at the top of the file. Could you move this up before functions?